sdk/provisioning/Azure.Provisioning/src/Parameter.cs

```diff
@@ -82,13 +93,29 @@ internal Parameter(string name, string? description, object? defaultValue, bool
         /// <param name="description">The parameter description.</param>
         /// <param name="defaultValue">The parameter defaultValue.</param>
         /// <param name="isSecure">Is the parameter secure.</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public Parameter(string name, string? description = default, object? defaultValue = default, bool isSecure = false)
+            : this(name, BicepKind.String, description, defaultValue, isSecure)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Parameter"/>.
+        /// </summary>
+        /// <param name="name">The parameter name.</param>
+        /// <param name="kind">The parameter kind.</param>
+        /// <param name="description">The parameter description.</param>
+        /// <param name="defaultValue">The parameter defaultValue.</param>
+        /// <param name="isSecure">Is the parameter secure.</param>
+        public Parameter(string name, BicepKind kind, string? description = default, object? defaultValue = default, bool isSecure = false)
```

Comment:
Should we default `kind` to `String`?  Then people could skip everything but the name.